### PR TITLE
Support requeuing proposals after a merge failure

### PR DESCRIPTION
**Pull request by bz2**
Look for reports that the merge failed when seeing if a pull request is
mergeable, and obey merge directives following it. The previous scheme
required all users to be admins to remove the 'merge request accepted'
messages from the bot to get it to retry.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/jenkins-github-lander/29/

**Comment by axw** [comment]
LGTM

File: src/jenkinsgithublander/github.py [hunk 1]
```diff
@@ -59,10 +60,14 @@ def _is_mergeable(comments, owner, trigger, request_info):
         if MERGE_SCHEDULED in comment['body']:
             is_merging = True
 
+        # Reset the status if a requested merge failed.
+        if is_merging and MERGE_FAILED in comment['body']:
```
Comment (mitechie) [inline src/jenkinsgithublander/github.py:64]: Hah, I was wondering how you guys were doing this in your setup.

**Comment by mitechie** [comment]
:+1: :shipit: i

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/jenkins-github-lander-merge

**Pull request merged**